src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -150,30 +150,72 @@ static class Extensions
 	{
 		private const uint Wgs84SpatialReferenceId = 4326;
 
-		public static Geoposition ToGeoPosition(this Location location)
-			=> new Geoposition(
-				new Geocoordinate(
-					latitude: location.Latitude,
-					longitude: location.Longitude,
-					altitude: location.Altitude,
-					timestamp: FromUnixTime(location.Time),
-					speed: location.HasSpeed ? location.Speed : 0,
-					point: new Geopoint(
-						new BasicGeoposition
-						{
-							Latitude = location.Latitude,
-							Longitude = location.Longitude,
-							Altitude = location.Altitude,
-						},
-						AltitudeReferenceSystem.Ellipsoid,
-						Wgs84SpatialReferenceId
-					),
-					accuracy: 0,
-					altitudeAccuracy: 0,
-					heading: null
-				)
-			);
-
+		public static Windows.Devices.Geolocation.Geoposition ToGeoPosition(this Location location)
+		{
+			double? geoheading = null;
+			if (location.HasBearing)
+			{
+				geoheading = location.Bearing;
+			}
+
+			Windows.Devices.Geolocation.PositionSource posSource;
+			switch (location.Provider)
+			{
+				case Android.Locations.LocationManager.NetworkProvider:
+					posSource = Windows.Devices.Geolocation.PositionSource.Cellular;    // cell, wifi
+					break;
+				case Android.Locations.LocationManager.PassiveProvider:
+					posSource = Windows.Devices.Geolocation.PositionSource.Unknown;  // other apps
+					break;
+				case Android.Locations.LocationManager.GpsProvider:
+					posSource = Windows.Devices.Geolocation.PositionSource.Satellite;
+					break;
+				default:
+					// ex.: "fused" - all merged, also e.g. Google Play
+					posSource = Windows.Devices.Geolocation.PositionSource.Unknown;
+					break;
+			}
+
+			Windows.Devices.Geolocation.BasicGeoposition basicGeoposition;

```

Comment:
Can you please put this back inline as it was before? There's no reason to explicitly declare it. Changes like this are 'noise' that makes it harder to see the actual functional changes that are relevant to the PR.